### ISSUE TITLE
Bug/Feature Request: Breakout of Style/AccessModifierDeclaration false positives (#5953)

### ISSUE THREAD

**Issue by luke-hill** [rubocop/rubocop#11910]
This is a breakout / re-opening of the old ticket. Specifically to fix the "main" use issue. (Non `def` defined methods shouldn't trip this up).

## Expected behavior

None `def` defined methods should not trip this up. As an example the below code should be permissible - as it's the most common style

```
class Foo
  attr_reader :baz
  private :baz

  def initialize(baz)
    @baz = baz
  end

  private

  # ten_other_methods
end
```

## Actual behavior

Same as prior to #5953 (Rubocop 0.x series)

## Steps to reproduce the problem

Write the code as provided above

## RuboCop version

Latest rubocop

**Comment by luke-hill** [comment]
EDIT: This does seem fixed in the latest version of rubocop. But not from the OP. So I'm going to close this as completed "at some point". I tried using rubocop 1.50+ and the cop offense isn't flagging for the above code example.

Tagging @koic for visibility

**Comment by koic** [comment]
@Vagab Yes, the issue is resolved on the master branch, but it hasn’t been released yet.

**Comment by luke-hill** [comment]
Hi @koic - I've seen this regression. What is the reference PR or note. Just because it's quite tricky to track stuff in rubocop. That way I know what to check for.

**Comment by Earlopain** [comment]
This is a consequence of https://github.com/rubocop/rubocop-ast/pull/325 and fixed by https://github.com/rubocop/rubocop/pull/13404 (I think)

**Comment by dvandersluis** [comment]
#13457 will have a similar effect FYI. Previously the cop was unable to properly process a symbol list or attr_* method with multiple items. If you have `AllowModifiersOnSymbols`/`AllowModifiersOnAttrs` set to `false`, you may get new offenses once released. Alternately, code that was previously flagged despite `AllowModifiersOnSymbols`/`AllowModifiersOnAttrs` being set to `true` no longer will.

**Comment by luke-hill** [comment]
I've not made any changes to anything on my repo. So from my POV I would ideally not like anything to break. It's quite a bit of extra checking I'm making. I'm not sure what I need to do / check in terms of next steps.

**Comment by Earlopain** [comment]
Only if you have set `AllowModifiersOnSymbols` or `AllowModifiersOnAttrs` to false will you get intended new offenses. The ones you currently get with the default config are considered a bug and will be fixed.
